test/integ/routers/alpha-router/alpha-router.integration.test.ts

```diff
@@ -0,0 +1,1021 @@
+/**
+ * @jest-environment hardhat
+ */
+
+import {
+  Currency,
+  CurrencyAmount,
+  Ether,
+  Percent,
+  Token,
+  TradeType,
+} from '@uniswap/sdk-core';
+import {
+  AlphaRouter,
+  AlphaRouterConfig,
+  ChainId,
+  DAI_MAINNET,
+  DAI_ON,
+  ID_TO_NETWORK_NAME,
+  ID_TO_PROVIDER,
+  nativeOnChain,
+  NATIVE_CURRENCY,
+  parseAmount,
+  SUPPORTED_CHAINS,
+  UniswapMulticallProvider,
+  UNI_GÖRLI,
+  UNI_MAINNET,
+  USDC_MAINNET,
+  USDC_ON,
+  USDT_MAINNET,
+  WETH9,
+  WNATIVE_ON,
+} from '../../../../src';
+
+import 'jest-environment-hardhat';
+
+import { JsonRpcProvider, JsonRpcSigner } from '@ethersproject/providers';
+
+import { Protocol } from '@uniswap/router-sdk';
+import { MethodParameters } from '@uniswap/v3-sdk';
+import { BigNumber, providers } from 'ethers';
+import { parseEther } from 'ethers/lib/utils';
+import _ from 'lodash';
+import { StaticGasPriceProvider } from '../../../../src/providers/static-gas-price-provider';
+import { DEFAULT_ROUTING_CONFIG_BY_CHAIN } from '../../../../src/routers/alpha-router/config';
+import { getBalanceAndApprove } from '../../../test-util/getBalanceAndApprove';
+
+const SWAP_ROUTER_V2 = '0x68b3465833fb72A70ecDF485E0e4C7bD8665Fc45';
+const SLIPPAGE = new Percent(5, 100); // 5% or 10_000?
+
+const checkQuoteToken = (
+  before: CurrencyAmount<Currency>,
+  after: CurrencyAmount<Currency>,
+  tokensQuoted: CurrencyAmount<Currency>
+) => {
+  // Check which is bigger to support exactIn and exactOut
+  const tokensSwapped = after.greaterThan(before)
+    ? after.subtract(before)
+    : before.subtract(after);
+  const tokensDiff = tokensQuoted.greaterThan(tokensSwapped)
+    ? tokensQuoted.subtract(tokensSwapped)
+    : tokensSwapped.subtract(tokensQuoted);
+  const percentDiff = tokensDiff.asFraction.divide(tokensQuoted.asFraction);
+  expect(percentDiff.lessThan(SLIPPAGE)).toBe(true);
+};
+
+const getQuoteToken = (
+  tokenIn: Currency,
+  tokenOut: Currency,
+  tradeType: TradeType
+): Currency => {
+  return tradeType == TradeType.EXACT_INPUT ? tokenOut : tokenIn;
+};
+
+export function parseDeadline(deadline: number): number {
+  return Math.floor(Date.now() / 1000) + deadline;
+}
+
+/// @dev had trouble with the return type casting to BigintIsh, wonder if there are overflow issues here
+const expandDecimals = (currency: Currency, amount: number): number => {
+  return amount * 10 ** currency.decimals;
+};
```

Comment:
I had issues casting the return of this func to `BigintIsh`, which is what is required by `CurrencyAmount.fromRawAmount(<Currency>, BigintIsh)`. This could overflow if somehow we have a huge number * 10 ** 18, but this is a testing env so I don't think it's too big of an issue